chain-tx-enclave/tx-validation/enclave/src/lib.rs

```diff
@@ -30,14 +30,17 @@ pub extern "C" fn ecall_initchain(chain_hex_id: u8) -> sgx_status_t {
     }
 }
 
+/// # Safety
+///
+/// This function should not be called with null pointer.
 #[no_mangle]
-pub extern "C" fn ecall_check_tx(
+pub unsafe extern "C" fn ecall_check_tx(
```

Comment:
does clippy suggest moving the unsafe?

Reply:
Yeah, clippy suggest `ecall_check_tx` itself should be unsafe.

Reply:
it'd be preferable to limit the unsafe scope... I'm thinking if `unsafe { slice::from_raw_parts(tx_request, tx_request_len).to_vec() }` would help?

Reply:
> it'd be preferable to limit the unsafe scope... I'm thinking if `unsafe { slice::from_raw_parts(tx_request, tx_request_len).to_vec() }` would help?

clippy says the pointer argument get dereferenced directly, maybe add a null pointer check is helpful.

Reply:
```
error: this public function dereferences a raw pointer but is not marked `unsafe`
  --> chain-tx-enclave/tx-validation/enclave/src/lib.rs:43:63
   |
43 |     let mut tx_request_slice = unsafe { slice::from_raw_parts(tx_request, tx_request_len) };
   |                                                               ^^^^^^^^^^
   |
   = note: `#[deny(clippy::not_unsafe_ptr_arg_deref)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#not_unsafe_ptr_arg_deref
```

Reply:
I guess it's indeed unsafe to dereference a arbitrary raw pointer.